src/coins.cpp

```diff
@@ -19,6 +23,39 @@ bool CCoinsView::HaveCoin(const COutPoint &outpoint) const
     return GetCoin(outpoint, coin);
 }
 
+bool CCoinsView::FindScriptPubKey(std::atomic<int>& scan_progress, const std::atomic<bool>& should_abort, int64_t& count, CCoinsViewCursor& cursor, const std::set<CScript>& needles, std::map<COutPoint, Coin>& out_results) {
+    scan_progress = 0;
+    count = 0;
+    while (cursor.Valid()) {
+        COutPoint key;
+        Coin coin;
+        if (!cursor.GetKey(key) || !cursor.GetValue(coin)) return false;
```

Comment:
This is refactored to a short-circuit in `Blockchain/RPC: Add scantxoutset method to scan UTXO set` after being added in `Add CCoinsView::FindScriptPubKey to search the UTXO set`. Instead, it should just be added refactored initially.